### DISCUSSION TITLE
Source of weather data

### DISCUSSION THREAD

**Discussion by joshcsmith13** [rocksdanister/weather#7] [Q&A] [answered]
I know I could dig through the code, but maybe you can tell me, and anybody else who is wondering, where does the weather data come from?

**Comment by rocksdanister** [comment]
<img width="874" alt="image" src="https://github.com/rocksdanister/weather/assets/17554161/5e9e35b3-6374-4961-9cc7-22d043ebf47e">

https://open-meteo.com

**Reply by rocksdanister** [reply, marked as answer]
Weather client implementation is abstracted:
https://github.com/rocksdanister/weather/tree/main/src/Drizzle.Weather

Other providers can be added easily without having to modify the app in other places.

**Reply by joshcsmith13** [reply]
Cool.  Thanks!  It looks like you have implemented interfaces for OpenMeteo and OpenWeatherMap.
